network/Message.h

```diff
@@ -139,22 +142,30 @@ class FO_COMMON_API Message {
     Message() = default;
     Message(MessageType message_type, std::string text);
 
-    MessageType        Type() const noexcept;      ///< Returns the type of the message.
-    std::size_t        Size() const noexcept;      ///< Returns the size of the underlying buffer.
-    const char*        Data() const noexcept;      ///< Returns the underlying buffer.
-    const std::string& Text() const;               ///< Returns the underlying buffer as a std::string.
+    MessageType        Type() const noexcept;       ///< Returns the type of the message.
+    std::size_t        Size() const noexcept;       ///< Returns the size of the underlying buffer.
+    bool               Compressed() const noexcept; ///< Returns the compression flag.
+    const char*        Data() const noexcept;       ///< Returns the underlying buffer.
+    const std::string& Text() const;                ///< Returns the underlying buffer as a std::string.
 
     void               Resize(std::size_t size);   ///< Resizes the underlying char buffer to \a size uninitialized bytes.
     char*              Data() noexcept;            ///< Returns the underlying buffer.
     void               Swap(Message& rhs) noexcept;///< Swaps the contents of \a *this with \a rhs.  Does not throw.
     void               Reset() noexcept;           ///< Reverts message to same state as after default constructor
 
+    void               Compress() noexcept;   ///< Compresses data, updates size and compression flag, if data is uncompressed.
```

Comment:
I am skeptical about `noexcept` given the large allocations

Reply:
Removed `noexcept` from `Compress()` and `Decompress()` in https://github.com/freeorion/freeorion/pull/3854/commits/4ead6795ce2b51bce839955a679750840ada4831.